### PR TITLE
Convert google provider and its tests to __future__.annotations

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -213,7 +213,9 @@ repos:
       - id: pyupgrade
         name: Upgrade Python code automatically
         args: ["--py37-plus"]
-        exclude: ^airflow/_vendor/
+        # We need to exclude gcs hook from pyupgrade because it has public "list" command which clashes
+        # with `list` that is used as type
```
Comment (ashb) [inline .pre-commit-config.yaml:217]: Is this a bug in pyupgrade?

**Review by ashb — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (potiuk) [reply .pre-commit-config.yaml:217]: Not really. This is a problem with naming our method "list". Simply Python thinks that "list" used as return type is our method.

Comment (potiuk) [reply .pre-commit-config.yaml:217]: Similarly as in the other case, I just let pyupgrade do the job and reverted the list change

Comment (potiuk) [reply .pre-commit-config.yaml:217]: Similarly as in the other case, I just let pyupgrade do the job and reverted the list change

**Comment by potiuk** [comment]
Closing as this one will be merged as part of #26289 #26290. See https://lists.apache.org/thread/81fr042s5d3v17v83bpo24tnrr2pp0fp for details

**Pull request closed without merging**